### PR TITLE
Send other claim instead of own

### LINKED ISSUE #40: meetup participants send other's claim instead of own
This issue is for the task **B1** from https://github.com/encointer/encointer-wallet-flutter/issues/113 that regard the pallet.

It involves:
* Extend the `ClaimOfAttendance` with a signature over itself
* `AttestationRegistry` should contain `attestees` instead of `attesters`
* adjust issue funds logic here, where it retrives the attestations from the registry: https://github.com/encointer/pallets/blob/95c8729bc0a8762272a52fe6372b4a2d307d7329/ceremonies/src/lib.rs#L529-L557
* adjust `register_attestations` xt
  * argument should be `Vec<ClaimOfAttendanceSigned>`
  * some checks need to be updated

### PR DESCRIPTION
**Pull request by clangenb**
* [ceremonies] instead of registering attestations. A meetup participant sends the other meetup participants' `ClaimOfAttendance`s to the chain, implicitly attesting their attendance. Hence, the `register_attestations` signature was changed to `attest_claims(origin, claims: Vec<ClaimOfAttendances>)`. Closes #40.
* To prevent forging of claims, the `ClaimOfAttendance` now contains a signature.
* The `Attestation` struct was removed because it was replaced with the signed `ClaimOfAttendance`

Tested with the encointer-node: `bootstrap_demo_currency` and `bot-community` run successfully.

Note: Merge #39 first.

### CODE REVIEW

File: ceremonies/src/lib.rs [hunk 1]
```diff
@@ -172,14 +172,14 @@ decl_module! {
...
             debug::RuntimeLogger::init();
             let sender = ensure_signed(origin)?;
             ensure!(<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::ATTESTING,
-                "registering attestations can only be done during ATTESTING phase");
+                "registering claims can only be done during ATTESTING phase");
             let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
-            ensure!(!attestations.is_empty(), "empty attestations supplied");
-            let cid = attestations[0].claim.community_identifier;
+            ensure!(!claims.is_empty(), "empty claims supplied");
+            let cid = claims[0].community_identifier;
```
Comment (clangenb) [inline ceremonies/src/lib.rs:182]: I have just come across this one.  This is potentially a new threat. If one claimant provides a wrong cid, all other claims are invalidated. Or shall we handle this in app? The claims are transparently stored as `ClaimOfAttendances`. We could easily discard invalid claims.

File: ceremonies/src/lib.rs [hunk 2]
```diff
@@ -531,39 +521,43 @@ impl<T: Config> Module<T> {
...
                             p
                         );
                         continue;
                     }
-                    let mut has_attested = 0u32;
-                    for w in attestations {
-                        let w_attestations = Self::attestation_registry(
+
+                    let mut was_attested_count = 0u32;
+                    for other_participant in &meetup_participants {
```
Comment (brenzi) [inline ceremonies/src/lib.rs:538]: so, the price for improved UX is one more for loop over each meetup. At least it's a +- linear increase in complexity for meetups of <=12

**Review by brenzi — APPROVED** [review]
love it.

File: ceremonies/src/lib.rs [hunk 3]
```diff
@@ -172,14 +172,14 @@ decl_module! {
         }
 
         #[weight = 10_000]
-        pub fn register_attestations(origin, attestations: Vec<Attestation<T::Signature, T::AccountId, T::Moment>>) -> DispatchResult {
+        pub fn attest_claims(origin, claims: Vec<ClaimOfAttendance<T::Signature, T::AccountId, T::Moment>>) -> DispatchResult {
             debug::RuntimeLogger::init();
             let sender = ensure_signed(origin)?;
             ensure!(<encointer_scheduler::Module<T>>::current_phase() == CeremonyPhaseType::ATTESTING,
-                "registering attestations can only be done during ATTESTING phase");
+                "registering claims can only be done during ATTESTING phase");
```
[the remark this answers is not in the document]
Comment (brenzi) [reply ceremonies/src/lib.rs:179]: I won't keep this PR for that matter....

**Pull request merged**